websites/M/Moopa/presence.ts

```diff
@@ -0,0 +1,234 @@
+							?.getAttribute("data-title-native") ||
+						document
+							.querySelector('meta[name="title"]')
+							?.getAttribute("data-title-romaji");
+
+					break;
+				}
+			}
+
+			presenceData.largeImageKey =
+				document
+					.querySelector('img[alt="poster anime"]')
+					?.getAttribute("src") || Assets.Logo;
+			presenceData.details = title || "Getting data...";
+			presenceData.state = "Viewing info";
+
+			if (title) {
+				presenceData.buttons = [
+					{
+						label: "View Anime",
+						url: href,
+					},
+				];
+			}
+		}
+	} else if (
+		pathname.includes("/manga/") &&
+		!pathname.includes("/read/") &&
+		!pathname.includes("/search/")
+	) {
+		if (document.querySelector('img[alt="404"]')) {
+			presenceData.state = "404 | Not Found";
+			presenceData.details = "Browsing Anime/Manga";
+		} else {
+			delete presenceData.startTimestamp;
+			const title = document.querySelector("h1.title")?.textContent;
+			presenceData.largeImageKey =
+				document.querySelector('img[alt="cover image"]')?.getAttribute("src") ||
+				Assets.Logo;
+			presenceData.details = title || "Getting data...";
+			presenceData.state = "Viewing info";
+
+			if (title) {
+				presenceData.buttons = [
+					{
+						label: "View Manga",
+						url: href,
+					},
+				];
+			}
+		}
+	} else if (pathname.includes("/watch/")) {
+		if (document.querySelector('img[alt="404"]')) {
+			presenceData.state = "404 | Not Found";
+			presenceData.details = "Browsing Anime/Manga";
+		} else {
+			const playingEpisode = document
+					.querySelector("h3")
+					?.textContent?.replace("Episode ", ""),
+				video: HTMLVideoElement = document.querySelector("video");
+			if (video && !Number.isNaN(Number(video.duration))) {
+				presenceData.endTimestamp = presence
+					.getTimestampsfromMedia(video)
+					.pop();
+				if (video.paused) {
+					presenceData.smallImageText = "Paused";
+					presenceData.smallImageKey = Assets.Pause;
+					delete presenceData.startTimestamp;
+					delete presenceData.endTimestamp;
+				} else {
+					presenceData.smallImageText = "Playing";
+					presenceData.smallImageKey = Assets.Play;
+				}
+			}
+
+			const total = document
+				.querySelector("div.grid.w-full.pl-5")
+				?.getAttribute("data-episode");
+
+			let title;
+			switch (titleLang) {
+				case 0: {
+					title = document
+						.querySelector('meta[name="title"]')
+						?.getAttribute("data-title-romaji");
+
+					break;
+				}
+				case 1: {
+					title =
+						document
+							.querySelector('meta[name="title"]')
+							?.getAttribute("data-title-english") ||
+						document
+							.querySelector('meta[name="title"]')
+							?.getAttribute("data-title-romaji");
+
+					break;
+				}
+				case 2: {
+					title =
+						document
+							.querySelector('meta[name="title"]')
+							?.getAttribute("data-title-native") ||
+						document
+							.querySelector('meta[name="title"]')
+							?.getAttribute("data-title-romaji");
+
+					break;
+				}
+			}
```

Comment:
This chunk of code is pretty large and repeated, it could be better as its own function.